### PR TITLE
Fix dialog body to be properly expanded

### LINKED ISSUE #4858: Merge column list is cut off in scroll in Firefox
Hi, ref: S/B 6084281

If user is on a laptop with small resolution and tries to merge a dataset that has >20 columns, they are not able to see all column names to add in Firefox. The scroll bar reaches the bottom of the page, but it has not reached the bottom of the column list.

I replicated this way:
1. From inside an existing dataset choose Edit > Merge with dataset
2. Column join
3. Choose this dataset for Step 3 of 3/ "Choose the rest to add":
https://team.cartodb.com/u/stephaniemongon/tables/commareas_chicago

This screenshot below is from Firefox 39.0, at 1024x640 resolution. Bottom of scrollbar properly shows last column names in other browsers like Chrome 44.0.2403.125

screen shot 2015-08-03 at 2 27 34 pm

### PR DESCRIPTION
**Pull request by viddo**
Fixes #4858 

Moved the rule, to add additional padding due to fixed footer, to an inner element instead as explained in http://stackoverflow.com/questions/26888428/display-flex-loses-right-padding-when-overflowing

Fixed the other places where this rules were necessary to change too

### CODE REVIEW

**Comment by viddo** [comment]
@xavijam please review

File: app/assets/stylesheets/common/dialog.css.scss [hunk 1]
```diff
@@ -244,15 +244,11 @@ $sStickyHeight: 97px;
...
-.Dialog-body--expanded.is-expandedWithSubFooter {
-  padding-bottom: $sStickyHeight;
-}
-.Dialog-body--expanded.is-expandedWithPreFooter {
-  padding-bottom: $sStickyHeight + 100px;
-}
+// These additional paddings are required to be applied on an inner element, for the paddings to apply in FF
+// See http://stackoverflow.com/questions/26888428/display-flex-loses-right-padding-when-overflowing for an isolated explanation
+.Dialog-bodyInnerExpandedWithSubFooter { padding-bottom: $sStickyHeight }
+.Dialog-bodyInnerExpandedWithPreFooter, { padding-bottom: $sStickyHeight + 100px }
```
Comment (xavijam) [inline app/assets/stylesheets/common/dialog.css.scss]: Be careful with the ```,``` before the ```{```

**Comment by xavijam** [comment]
Go for it man!

File: app/assets/stylesheets/common/dialog.css.scss (hunk 1 above)
Comment (viddo) [inline app/assets/stylesheets/common/dialog.css.scss]: Ouch, thx

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/5633/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/5634/))

**Pull request merged**